src/workerd/io/worker.c++

```diff
@@ -2182,15 +2219,28 @@ void Worker::Lock::validateHandlers(ValidationErrorReporter& errorReporter) {
           //   hence we will see it here. Rather than try to correct this inconsistency between
           //   struct and dict handling (which could have unintended consequences), let's just
           //   work around by ignoring arrays here.
+          errorReporter.addEntrypoint(name, kj::Array<kj::String>());
         } else {
+          // Use a HashSet to avoid duplicates when methods exist both as own properties
+          // and in the prototype chain
+          kj::HashSet<kj::String> methodSet;
+
+          // First, check for own properties (like a plain object literal)
           auto dict = js.toDict(handle);
           for (auto& field: dict.fields) {
             if (!ignoredHandlers.contains(field.name)) {
-              methods.add(kj::mv(field.name));
+              methodSet.upsert(kj::mv(field.name), [&](auto&, auto&&) {});
             }
           }
+
+          // Then, check for methods in the prototype chain (like a class instance)
+          js.withinHandleScope([&]() {
+            collectMethodsFromPrototypeChain(jsg::JsObject(handle).getPrototype(js), methodSet);
```

Comment:
So this is the meaningful line change in this PR, yeah? Previously you weren't walking the prototype chain for default exports, is that right? I guess I'm not super clear why default is special, I suppose historical reasons?

I presume `worker.impl->statelessClasses` includes both `WorkerEntrypoint` and `ExportedHandler` style exports, right? Or is `ExportedHandler` handled elsewhere?

Reply:
`statelessClasses` is strictly `WorkerEntrypoint`. `ExportedHandler` style is what we're handling here.

Reply:
To be clear this code here is used for all plain-object exports, whether default or not.